src/hotspot/share/utilities/elfFile.cpp

```diff
@@ -1624,6 +1628,29 @@ bool DwarfFile::LineNumberProgram::get_filename_from_header(const uint32_t file_
   return false;
 }
 
+// Remove everything before the last slash including the slash itself to get the actual filename. This is required, for
+// example, for Clang debug builds which emit a relative path while GCC only emits the filename.
+void DwarfFile::LineNumberProgram::strip_path_prefix(char* filename, const size_t filename_len) {
+  char* last_slash = strrchr(filename, *os::file_separator());
+  if (last_slash != nullptr) {
+    uint16_t index_after_slash = (uint16_t)(last_slash + 1 - filename);
+    // Copy filename to beginning of buffer.
+    int bytes_written = jio_snprintf(filename, filename_len - index_after_slash, "%s", filename + index_after_slash);
+    assert(bytes_written > 0, "could not strip path prefix");
+    // Add null terminator.
+    filename[bytes_written] = '\0';
+  }
+}
+
+// Copy the read filename from the DWARF file stored in 'src' to the provided output buffer 'dst'.
+void DwarfFile::LineNumberProgram::copy_dwarf_filename_to_filename(char* src, const size_t src_len,
+                                                                   char* dst, const size_t dst_len) {
+  const size_t count = MIN(src_len, dst_len);
+  int bytes_written = jio_snprintf(dst, count, "%s", src);
+  // Add null terminator.
+  dst[count - 1] = '\0';
```

Comment:
Does it make sense to return a truncated file name up to the caller of `DwarfFile::LineNumberProgram::get_filename_from_header()`? Will this not just confuse him? I think it makes more sense to cleanly handle truncation, and e.g. skip file parsing for dwarf files with too long names.

Reply:
I think you're right that it's rather unexpected to get an incomplete filename back. But just silently skipping the filename might be confusing as well. We could either print an error (I guess that's useful either way but should only be printed with `TraceDwarfLevel`) or just return a generic "buffer overflow" string as filename instead if it fits into the provided filename buffer. And only if that's not possible we could silently skip the filename. Would that be an option?

Reply:
Optional tracing and returning a generic string sounds fine.